lib/ast.js

```diff
@@ -589,61 +589,64 @@ var AST_Seq = DEFNODE("Seq", "car cdr", {
         cdr: "[AST_Node] second element in sequence"
     },
     $cons: function(x, y) {
+        if (!x) {
+            throw new Error('AST_Seq.car missing');
+        }
```

Comment:
I am not in favor of checks in these internal functions that would slow down uglify - otherwise we could have checks everywhere for incompatible inputs. I think documenting current behavior in comments is sufficient.